add search ionames in different dirs\n set -ionames by default

remixoff opened the pull request:
* add search stm8.inc in different dirs (including current)
* set -ionames behavior by default (and swap parameter with reverse 'noionames')

Commit:
add search ionames in different dirs\n set -ionames by default

volbus commented:
In the proposed change, with multiple file locations for the ionames, if try_open fails will cause a failure program exit. In the current version, a file open failure will simply switch off the usage of  IO-names, using addresses by default.
And about the default mode to use IO register names, I will not accept it, because the current file version for register names is not really complete and there are still some missing definitions - especially for bit locations. Also, due to the fact, that the IO names file can be edited to add/remove definitions, errors can be added, so this should not be the default mode. Default mode will write clean addresses.

remixoff commented:
> In the current version, a file open failure will simply switch off the usage of IO-names, using addresses by default. 

I could actually disappoint you, but not ) 
Because on the line [gmtdisas.c:92](https://github.com/volbus/gmtdisas/blob/main/gmtdisas.c#L92) you use iofile variable even if fopen on [gmtdisas.c:81](https://github.com/volbus/gmtdisas/blob/main/gmtdisas.c#L81) returned zero. Which is a null-pointer exception. )

I improved it too for now, and fixed my own bugs I tried to add :-[

> And about the default mode to use IO register names, I will not accept it, because the current file version for register names is not really complete and there are still some missing definitions - especially for bit locations. Also, due to the fact, that the IO names file can be edited to add/remove definitions, errors can be added, so this should not be the default mode. Default mode will write clean addresses.

Actually, it's sad from my POV, but you are the owner. I reverted it back. 

PS: From my POV it's any way non-guarantee, but my story: I just cloned it, says 'thanks' to you, and disassembled one small FW. Then I spent some time for figure out. Only after that I read the help ) This why I thought that default mode with symbols could be reasonable.

schneidersoft commented:
I agree with volbus that the default should be robust and should not depend on external configuration files.
